### ISSUE TITLE
Failed to start MPP Solar Service

### ISSUE THREAD

**Issue by dricote** [jblance/mpp-solar#16]
Hello, I cannot start the service, I use a raspi2 and Raspbian Buster, when I enter this I get the following error.

```
pi@PIP5048MG:~ $ systemctl --user status mpp-solar
● mpp-solar.service - MPP Solar Service
   Loaded: loaded (/etc/systemd/user/mpp-solar.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sun 2020-04-26 06:10:55 CEST; 2min 23s ago
  Process: 789 ExecStart=/usr/bin/python /usr/local/bin/mpp-solar-service -c /etc/mpp-solar/m
 Main PID: 789 (code=exited, status=1/FAILURE)

abr 26 06:10:55 PIP5048MG systemd[542]: Failed to start MPP Solar Service.
abr 26 06:10:55 PIP5048MG systemd[542]: mpp-solar.service: Service RestartSec=100ms expired,
abr 26 06:10:55 PIP5048MG systemd[542]: mpp-solar.service: Scheduled restart job, restart cou
abr 26 06:10:55 PIP5048MG systemd[542]: Stopped MPP Solar Service.
abr 26 06:10:55 PIP5048MG systemd[542]: mpp-solar.service: Start request repeated too quickly
abr 26 06:10:55 PIP5048MG systemd[542]: mpp-solar.service: Failed with result 'exit-code'.
abr 26 06:10:55 PIP5048MG systemd[542]: Failed to start MPP Solar Service.
```
Thanks

**Comment by dricote** [comment]
This is my mpp-solar.conf

```
[SETUP]
# Number of seconds to pause between command execution loop
# 0 is no pause, greater than 60 will cause service restarts
pause=5
mqtt_broker=localhost

# This example would work on 2x PIP4048 installed in parallel
[Inverter_1]
model=standard
port=/dev/hidraw0
baud=2400
command=QPGS0
tag=QPGS0
format=influx2
```

**Comment by jblance** [comment]
Hi
try running the service command on the command line 
`/usr/bin/python /usr/local/bin/mpp-solar-service -c /etc/mpp-solar/mpp-solar.conf`
next try the command line version:
`mpp-info-pub -q localhost -d /dev/hidraw0 -c QPGS0 -I`
finally try the standard mpp-solar command (which allows extra info)
`mpp-solar -d /dev/hidraw0 -c QPGS0 -I` (change `-I` to `-D` for even more info on this command)
post any output

Cheers

**Comment by dricote** [comment]
When trying to lift the service it tells me the following
/usr/bin/python /usr/local/bin/mpp-solar-service -c /etc/mpp-solar/mpp-solar.conf


```
pi@PIP5048MG:~ $ /usr/bin/python /usr/local/bin/mpp-solar-service -c /etc/mpp-solar/mpp-solar.conf
Traceback (most recent call last):
  File "/usr/local/bin/mpp-solar-service", line 11, in <module>
    load_entry_point('mpp-solar==0.3.1', 'console_scripts', 'mpp-solar-service')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2793, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2411, in load
    return self.resolve()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2417, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/mpp_solar_service.py", line 11, in <module>
    import paho.mqtt.publish as publish
ImportError: No module named paho.mqtt.publish
```

**Comment by dricote** [comment]
Seen the error it gave, I have looked at another issue and installed paho-mqtt with `sudo pip install paho-mqtt` and the result has been the following:

```
pi@PIP5048MG:~ $ /usr/bin/python /usr/local/bin/mpp-solar-service -c /etc/mpp-solar/mpp-solar.conf
MPP-Solar-Service: Initializing ...
MPP-Solar-Service: Config file: /etc/mpp-solar/mpp-solar.conf
MPP-Solar-Service: Config setting - pause: 5
MPP-Solar-Service: Config setting - mqtt_broker: localhost
MPP-Solar-Service: Config setting - command sections found: 1
MPP-Solar-Service: item {'tag': u'QPGS0', 'command': u'QPGS0', 'mp': <mppsolar.mpputils.mppUtils instance at 0x75dd45f8>, 'format': u'influx2'}
```

Then I have activated the service normally with `systemctl --user start mpp-solar`

```
pi@PIP5048MG:~ $ systemctl --user start mpp-solar
pi@PIP5048MG:~ $ systemctl --user status mpp-solar
● mpp-solar.service - MPP Solar Service
   Loaded: loaded (/etc/systemd/user/mpp-solar.service; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2020-04-26 18:27:44 CEST; 9s ago
 Main PID: 1532 (python)
   CGroup: /user.slice/user-1000.slice/user@1000.service/mpp-solar.service
           └─1532 /usr/bin/python /usr/local/bin/mpp-solar-service -c /etc/mpp-solar/mpp-solar.

abr 26 18:27:42 PIP5048MG systemd[413]: Starting MPP Solar Service...
abr 26 18:27:44 PIP5048MG python[1532]: MPP-Solar-Service: Initializing ...
abr 26 18:27:44 PIP5048MG python[1532]: MPP-Solar-Service: Config file: /etc/mpp-solar/mpp-sola
abr 26 18:27:44 PIP5048MG python[1532]: MPP-Solar-Service: Config setting - pause: 5
abr 26 18:27:44 PIP5048MG python[1532]: MPP-Solar-Service: Config setting - mqtt_broker: localh
abr 26 18:27:44 PIP5048MG python[1532]: MPP-Solar-Service: Config setting - command sections fo
abr 26 18:27:44 PIP5048MG python[1532]: MPP-Solar-Service: item {'tag': u'QPGS0', 'command': u'
abr 26 18:27:44 PIP5048MG systemd[413]: Started MPP Solar Service.
```
I had some packages retained when upgrade that I do not know if they would have to see, they are the following:
`binutils binutils-arm-linux-gnueabihf binutils-common libbinutils`
I have installed them and then I have done the installation of `paho-mqtt`

Then i tried `mpp-info-pub -q localhost -d /dev/hidraw0 -c QPGS0 -I` with `sudo` in front because otherwise it doesn't work for me, and in another console I have received the following:

```
pi@PIP5048MG:~ $ mosquitto_sub -h localhost -v -t "#"
mpp-solar mpp-solar,command=QPGS0 parallel_instance_number="valid"
mpp-solar mpp-solar,command=QPGS0 serial_number=9.29319031031e+13
mpp-solar mpp-solar,command=QPGS0 work_mode="Battery Mode"
mpp-solar mpp-solar,command=QPGS0 fault_code="No fault"
mpp-solar mpp-solar,command=QPGS0 grid_voltage=0.0
mpp-solar mpp-solar,command=QPGS0 grid_frequency=0.0
mpp-solar mpp-solar,command=QPGS0 ac_output_voltage=230.1
mpp-solar mpp-solar,command=QPGS0 ac_output_frequency=50.0
mpp-solar mpp-solar,command=QPGS0 ac_output_apparent_power=115.0
mpp-solar mpp-solar,command=QPGS0 ac_output_active_power=108.0
mpp-solar mpp-solar,command=QPGS0 load_percentage=2.0
mpp-solar mpp-solar,command=QPGS0 battery_voltage=53.3
mpp-solar mpp-solar,command=QPGS0 battery_charging_current=2.0
mpp-solar mpp-solar,command=QPGS0 battery_capacity=75.0
mpp-solar mpp-solar,command=QPGS0 pv_input_voltage=223.2
mpp-solar mpp-solar,command=QPGS0 total_charging_current=2.0
mpp-solar mpp-solar,command=QPGS0 total_ac_output_apparent_power=115.0
mpp-solar mpp-solar,command=QPGS0 total_output_active_power=108.0
mpp-solar mpp-solar,command=QPGS0 total_ac_output_percentage=2.0
mpp-solar mpp-solar,command=QPGS0 is_scc_ok=1
mpp-solar mpp-solar,command=QPGS0 is_ac_charging=0
mpp-solar mpp-solar,command=QPGS0 is_scc_charging=1
mpp-solar mpp-solar,command=QPGS0 is_battery_over_voltage=0
mpp-solar mpp-solar,command=QPGS0 is_battery_under_voltage=0
mpp-solar mpp-solar,command=QPGS0 is_line_lost=1
mpp-solar mpp-solar,command=QPGS0 is_load_on=1
mpp-solar mpp-solar,command=QPGS0 is_configuration_changed=0
mpp-solar mpp-solar,command=QPGS0 output_mode="single machine"
mpp-solar mpp-solar,command=QPGS0 charger_source_priority="Solar + Utility"
mpp-solar mpp-solar,command=QPGS0 max_charger_current=80.0
mpp-solar mpp-solar,command=QPGS0 max_charger_range=80.0
mpp-solar mpp-solar,command=QPGS0 max_ac_charger_current=50.0
mpp-solar mpp-solar,command=QPGS0 pv_input_current=2.0
mpp-solar mpp-solar,command=QPGS0 battery_discharge_current=0.0
```
finally I have done what you have told me

```
pi@PIP5048MG:~ $ sudo mpp-solar -d /dev/hidraw0 -c QPGS0 -I
INFO:MPP-Solar:command QPGS0
INFO:MPP-Solar:Serial device used: /dev/hidraw0, baud rate: 2400
INFO:MPP-Solar:Loading commands for inverter model: standard
INFO:MPP-Solar:... command F loaded for inverter model: standard
INFO:MPP-Solar:... command MCHGC loaded for inverter model: standard
INFO:MPP-Solar:... command MNCHGC loaded for inverter model: standard
INFO:MPP-Solar:... command MUCHGC loaded for inverter model: standard
INFO:MPP-Solar:... command PBCV loaded for inverter model: standard
INFO:MPP-Solar:... command PBDV loaded for inverter model: standard
INFO:MPP-Solar:... command PBFT loaded for inverter model: standard
INFO:MPP-Solar:... command PBT loaded for inverter model: standard
INFO:MPP-Solar:... command PCP loaded for inverter model: standard
INFO:MPP-Solar:... command PCVV loaded for inverter model: standard
INFO:MPP-Solar:... command PEPD loaded for inverter model: standard
INFO:MPP-Solar:... command PF loaded for inverter model: standard
INFO:MPP-Solar:... command PGR loaded for inverter model: standard
INFO:MPP-Solar:... command POP loaded for inverter model: standard
INFO:MPP-Solar:... command POPM loaded for inverter model: standard
INFO:MPP-Solar:... command PPCP loaded for inverter model: standard
INFO:MPP-Solar:... command PPVOKC loaded for inverter model: standard
INFO:MPP-Solar:... command PSDV loaded for inverter model: standard
INFO:MPP-Solar:... command PSPB loaded for inverter model: standard
INFO:MPP-Solar:... command Q1 loaded for inverter model: standard
INFO:MPP-Solar:... command QBOOT loaded for inverter model: standard
INFO:MPP-Solar:... command QDI loaded for inverter model: standard
INFO:MPP-Solar:... command QFLAG loaded for inverter model: standard
INFO:MPP-Solar:... command QID loaded for inverter model: standard
INFO:MPP-Solar:... command QMCHGCR loaded for inverter model: standard
INFO:MPP-Solar:... command QMOD loaded for inverter model: standard
INFO:MPP-Solar:... command QMUCHGCR loaded for inverter model: standard
INFO:MPP-Solar:... command QOPM loaded for inverter model: standard
INFO:MPP-Solar:... command QPGS loaded for inverter model: standard
INFO:MPP-Solar:... command QPI loaded for inverter model: standard
INFO:MPP-Solar:... command QPIGS loaded for inverter model: standard
INFO:MPP-Solar:... command QPIRI loaded for inverter model: standard
INFO:MPP-Solar:... command QPIWS loaded for inverter model: standard
INFO:MPP-Solar:... command QVFW loaded for inverter model: standard
INFO:MPP-Solar:... command QVFW2 loaded for inverter model: standard
INFO:MPP-Solar:DIRECT USB connection: executing QPGS
Parallel Information inquiry
 -- example: QPGS1 queries the values of various metrics from instance 1 of parallel setup Inverters (numbers from 0)

ac_output_active_power          0107            W
ac_output_apparent_power        0115            VA
ac_output_frequency             50.00           Hz
ac_output_voltage               230.0           V
battery_capacity                074             %
battery_charging_current        000             A
battery_discharge_current       000             A
battery_voltage                 53.2            V
charger_source_priority         Solar + Utility
fault_code                      No fault
grid_frequency                  00.00           Hz
grid_voltage                    000.0           V
is_ac_charging                  0               True - 1/False - 0
is_battery_over_voltage         0               True - 1/False - 0
is_battery_under_voltage        0               True - 1/False - 0
is_configuration_changed        0               True - 1/False - 0
is_line_lost                    1               True - 1/False - 0
is_load_on                      1               True - 1/False - 0
is_scc_charging                 1               True - 1/False - 0
is_scc_ok                       1               True - 1/False - 0
load_percentage                 002             %
max_ac_charger_current          50              A
max_charger_current             080             A
max_charger_range               080             A
output_mode                     single machine
parallel_instance_number        valid
pv_input_current                00              A
pv_input_voltage                220.6           V
serial_number                   92931903103094
total_ac_output_apparent_power  00115           VA
total_ac_output_percentage      002             %
total_charging_current          000             A
total_output_active_power       00107           W
work_mode                       Battery Mode
```
and with -D

```
pi@PIP5048MG:~ $ sudo mpp-solar -d /dev/hidraw0 -c QPGS0 -D
INFO:MPP-Solar:command QPGS0
INFO:MPP-Solar:Serial device used: /dev/hidraw0, baud rate: 2400
DEBUG:MPP-Solar:Device matches hidraw regex
INFO:MPP-Solar:Loading commands for inverter model: standard
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/f.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command F supported all inverters
INFO:MPP-Solar:... command F loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for F
DEBUG:MPP-Solar:Calculating CRC for F
DEBUG:MPP-Solar:Generated CRC 29 2 2902
DEBUG:MPP-Solar:Full command: F)
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/mchgc.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command MCHGC supported all inverters
INFO:MPP-Solar:... command MCHGC loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for MCHGC
DEBUG:MPP-Solar:Calculating CRC for MCHGC
DEBUG:MPP-Solar:Generated CRC 9d a2 9da2
DEBUG:MPP-Solar:Full command: MCHGC▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/mnchgc.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command MNCHGC supported all inverters
INFO:MPP-Solar:... command MNCHGC loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for MNCHGC
DEBUG:MPP-Solar:Calculating CRC for MNCHGC
DEBUG:MPP-Solar:Generated CRC 57 23 5723
DEBUG:MPP-Solar:Full command: MNCHGCW#
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/muchgc.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command MUCHGC supported all inverters
INFO:MPP-Solar:... command MUCHGC loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for MUCHGC
DEBUG:MPP-Solar:Calculating CRC for MUCHGC
DEBUG:MPP-Solar:Generated CRC bf 86 bf86
DEBUG:MPP-Solar:Full command: MUCHGC▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pbcv.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PBCV supported all inverters
INFO:MPP-Solar:... command PBCV loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PBCV
DEBUG:MPP-Solar:Calculating CRC for PBCV
DEBUG:MPP-Solar:Generated CRC 64 5a 645a
DEBUG:MPP-Solar:Full command: PBCVdZ
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pbdv.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PBDV supported all inverters
INFO:MPP-Solar:... command PBDV loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PBDV
DEBUG:MPP-Solar:Calculating CRC for PBDV
DEBUG:MPP-Solar:Generated CRC fd cd fdcd
DEBUG:MPP-Solar:Full command: PBDV▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pbft.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PBFT supported all inverters
INFO:MPP-Solar:... command PBFT loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PBFT
DEBUG:MPP-Solar:Calculating CRC for PBFT
DEBUG:MPP-Solar:Generated CRC bb ed bbed
DEBUG:MPP-Solar:Full command: PBFT▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pbt.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PBT supported all inverters
INFO:MPP-Solar:... command PBT loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PBT
DEBUG:MPP-Solar:Calculating CRC for PBT
DEBUG:MPP-Solar:Generated CRC 2f 11 2f11
DEBUG:MPP-Solar:Full command: PBT/
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pcp.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PCP supported all inverters
INFO:MPP-Solar:... command PCP loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PCP
DEBUG:MPP-Solar:Calculating CRC for PCP
DEBUG:MPP-Solar:Generated CRC 5c a4 5ca4
DEBUG:MPP-Solar:Full command: PCP\▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pcvv.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PCVV supported all inverters
INFO:MPP-Solar:... command PCVV loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PCVV
DEBUG:MPP-Solar:Calculating CRC for PCVV
DEBUG:MPP-Solar:Generated CRC af ec afec
DEBUG:MPP-Solar:Full command: PCVV▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pepd.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PEPD supported all inverters
INFO:MPP-Solar:... command PEPD loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PEPD
DEBUG:MPP-Solar:Calculating CRC for PEPD
DEBUG:MPP-Solar:Generated CRC 85 99 8599
DEBUG:MPP-Solar:Full command: PEPD▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pf.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PF supported all inverters
INFO:MPP-Solar:... command PF loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PF
DEBUG:MPP-Solar:Calculating CRC for PF
DEBUG:MPP-Solar:Generated CRC 26 bd 26bd
DEBUG:MPP-Solar:Full command: PF&▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pgr.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PGR supported all inverters
INFO:MPP-Solar:... command PGR loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PGR
DEBUG:MPP-Solar:Calculating CRC for PGR
DEBUG:MPP-Solar:Generated CRC b0 22 b022
DEBUG:MPP-Solar:Full command: PGR▒"
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pop.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command POP supported all inverters
INFO:MPP-Solar:... command POP loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for POP
DEBUG:MPP-Solar:Calculating CRC for POP
DEBUG:MPP-Solar:Generated CRC 19 c9 19c9
DEBUG:MPP-Solar:Full command: POP▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/popm.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command POPM supported all inverters
INFO:MPP-Solar:... command POPM loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for POPM
DEBUG:MPP-Solar:Calculating CRC for POPM
DEBUG:MPP-Solar:Generated CRC d3 71 d371
DEBUG:MPP-Solar:Full command: POPM▒q
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/ppcp.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PPCP supported all inverters
INFO:MPP-Solar:... command PPCP loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PPCP
DEBUG:MPP-Solar:Calculating CRC for PPCP
DEBUG:MPP-Solar:Generated CRC 29 9f 299f
DEBUG:MPP-Solar:Full command: PPCP)▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/ppvokc.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PPVOKC supported all inverters
INFO:MPP-Solar:... command PPVOKC loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PPVOKC
DEBUG:MPP-Solar:Calculating CRC for PPVOKC
DEBUG:MPP-Solar:Generated CRC af 9 af09
DEBUG:MPP-Solar:Full command: PPVOKC▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/psdv.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PSDV supported all inverters
INFO:MPP-Solar:... command PSDV loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PSDV
DEBUG:MPP-Solar:Calculating CRC for PSDV
DEBUG:MPP-Solar:Generated CRC 89 9e 899e
DEBUG:MPP-Solar:Full command: PSDV▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/pspb.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command PSPB supported all inverters
INFO:MPP-Solar:... command PSPB loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for PSPB
DEBUG:MPP-Solar:Calculating CRC for PSPB
DEBUG:MPP-Solar:Generated CRC 14 9c 149c
DEBUG:MPP-Solar:Full command: PSPB▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/q1.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command Q1 supported all inverters
INFO:MPP-Solar:... command Q1 loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for Q1
DEBUG:MPP-Solar:Calculating CRC for Q1
DEBUG:MPP-Solar:Generated CRC 1b fc 1bfc
DEBUG:MPP-Solar:Full command: Q1
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qboot.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QBOOT supported all inverters
INFO:MPP-Solar:... command QBOOT loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QBOOT
DEBUG:MPP-Solar:Calculating CRC for QBOOT
DEBUG:MPP-Solar:Generated CRC b 88 b88
DEBUG:MPP-Solar:Full command: QBOOT
                                   ▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qdi.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QDI supported all inverters
INFO:MPP-Solar:... command QDI loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QDI
DEBUG:MPP-Solar:Calculating CRC for QDI
DEBUG:MPP-Solar:Generated CRC 71 1b 711b
DEBUG:MPP-Solar:Full command: QDIq

EBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qflag.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QFLAG supported all inverters
INFO:MPP-Solar:... command QFLAG loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QFLAG
DEBUG:MPP-Solar:Calculating CRC for QFLAG
DEBUG:MPP-Solar:Generated CRC 98 74 9874
DEBUG:MPP-Solar:Full command: QFLAG▒t
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qid.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QID supported all inverters
INFO:MPP-Solar:... command QID loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QID
DEBUG:MPP-Solar:Calculating CRC for QID
DEBUG:MPP-Solar:Generated CRC d6 ea d6ea
DEBUG:MPP-Solar:Full command: QID▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qmchgcr.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QMCHGCR supported all inverters
INFO:MPP-Solar:... command QMCHGCR loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QMCHGCR
DEBUG:MPP-Solar:Calculating CRC for QMCHGCR
DEBUG:MPP-Solar:Generated CRC d8 55 d855
DEBUG:MPP-Solar:Full command: QMCHGCR▒U
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qmod.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QMOD supported all inverters
INFO:MPP-Solar:... command QMOD loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QMOD
DEBUG:MPP-Solar:Calculating CRC for QMOD
DEBUG:MPP-Solar:Generated CRC 49 c1 49c1
DEBUG:MPP-Solar:Full command: QMODI▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qmuchgcr.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QMUCHGCR supported all inverters
INFO:MPP-Solar:... command QMUCHGCR loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QMUCHGCR
DEBUG:MPP-Solar:Calculating CRC for QMUCHGCR
DEBUG:MPP-Solar:Generated CRC 26 34 2634
DEBUG:MPP-Solar:Full command: QMUCHGCR&4
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qopm.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QOPM supported all inverters
INFO:MPP-Solar:... command QOPM loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QOPM
DEBUG:MPP-Solar:Calculating CRC for QOPM
DEBUG:MPP-Solar:Generated CRC a5 c5 a5c5
DEBUG:MPP-Solar:Full command: QOPM▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qp2gs.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qpgs-lv.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qpgs.json
DEBUG:MPP-Solar:-----No supports [u'LV5048']
DEBUG:MPP-Solar:Command QPGS supported all inverters
INFO:MPP-Solar:... command QPGS loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QPGS
DEBUG:MPP-Solar:Calculating CRC for QPGS
DEBUG:MPP-Solar:Generated CRC a3 8c a38c
DEBUG:MPP-Solar:Full command: QPGS▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qpi.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QPI supported all inverters
INFO:MPP-Solar:... command QPI loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QPI
DEBUG:MPP-Solar:Calculating CRC for QPI
DEBUG:MPP-Solar:Generated CRC be ac beac
DEBUG:MPP-Solar:Full command: QPI▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qpigs-lv.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qpigs.json
DEBUG:MPP-Solar:-----No supports [u'LV5048']
DEBUG:MPP-Solar:Command QPIGS supported all inverters
INFO:MPP-Solar:... command QPIGS loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QPIGS
DEBUG:MPP-Solar:Calculating CRC for QPIGS
DEBUG:MPP-Solar:Generated CRC b7 a9 b7a9
DEBUG:MPP-Solar:Full command: QPIGS▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qpigs2.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qpiri-lv.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qpiri.json
DEBUG:MPP-Solar:-----No supports [u'LV5048']
DEBUG:MPP-Solar:Command QPIRI supported all inverters
INFO:MPP-Solar:... command QPIRI loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QPIRI
DEBUG:MPP-Solar:Calculating CRC for QPIRI
DEBUG:MPP-Solar:Generated CRC f8 54 f854
DEBUG:MPP-Solar:Full command: QPIRI▒T
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qpiws.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QPIWS supported all inverters
INFO:MPP-Solar:... command QPIWS loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QPIWS
DEBUG:MPP-Solar:Calculating CRC for QPIWS
DEBUG:MPP-Solar:Generated CRC b4 da b4da
DEBUG:MPP-Solar:Full command: QPIWS▒▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qvfw.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QVFW supported all inverters
INFO:MPP-Solar:... command QVFW loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QVFW
DEBUG:MPP-Solar:Calculating CRC for QVFW
DEBUG:MPP-Solar:Generated CRC 62 99 6299
DEBUG:MPP-Solar:Full command: QVFWb▒
DEBUG:MPP-Solar:Loading command information from /usr/local/lib/python2.7/dist-packages/mpp_solar-0.3.1-py2.7.egg/mppsolar/commands/qvfw2.json
DEBUG:MPP-Solar:-----No supports
DEBUG:MPP-Solar:Command QVFW2 supported all inverters
INFO:MPP-Solar:... command QVFW2 loaded for inverter model: standard
DEBUG:MPP-Solar:Generate full command for QVFW2
DEBUG:MPP-Solar:Calculating CRC for QVFW2
DEBUG:MPP-Solar:Generated CRC c3 f5 c3f5
DEBUG:MPP-Solar:Full command: QVFW2▒▒
DEBUG:MPP-Solar:Searching for cmd 'QPGS0'
DEBUG:MPP-Solar:Matched: QPGS Value: 0
DEBUG:MPP-Solar:Generate full command for QPGS0
DEBUG:MPP-Solar:Calculating CRC for QPGS0
DEBUG:MPP-Solar:Generated CRC 3f da 3fda
DEBUG:MPP-Solar:Full command: QPGS0?▒
INFO:MPP-Solar:DIRECT USB connection: executing QPGS
Parallel Information inquiry
 -- example: QPGS1 queries the values of various metrics from instance 1 of parallel setup Inverters (numbers from 0)

DEBUG:MPP-Solar:length of to_send: 8
DEBUG:MPP-Solar:1 chunk send
DEBUG:MPP-Solar:usb response was: (1 92931903103094 B 00 000.0 00.00 230.1 49.99 0115 0099 002 53.1 000 073 222.6 000 00115 00099 001 10100110 0 2 080 080 50 00 000.N
DEBUG:MPP-Solar:Response length: 133
DEBUG:MPP-Solar:CRC resp        2e 4e
DEBUG:MPP-Solar:Calculating CRC for (1 92931903103094 B 00 000.0 00.00 230.1 49.99 0115 0099 002 53.1 000 073 222.6 000 00115 00099 001 10100110 0 2 080 080 50 00 000
DEBUG:MPP-Solar:Generated CRC 2e 4e 2e4e
DEBUG:MPP-Solar:CRC calc        2e 4e
DEBUG:MPP-Solar:CRCs match
DEBUG:MPP-Solar:Response valid as no invalid situations found
ac_output_active_power          0099            W
ac_output_apparent_power        0115            VA
ac_output_frequency             49.99           Hz
ac_output_voltage               230.1           V
battery_capacity                073             %
battery_charging_current        000             A
battery_discharge_current       000             A
battery_voltage                 53.1            V
charger_source_priority         Solar + Utility
fault_code                      No fault
grid_frequency                  00.00           Hz
grid_voltage                    000.0           V
is_ac_charging                  0               True - 1/False - 0
is_battery_over_voltage         0               True - 1/False - 0
is_battery_under_voltage        0               True - 1/False - 0
is_configuration_changed        0               True - 1/False - 0
is_line_lost                    1               True - 1/False - 0
is_load_on                      1               True - 1/False - 0
is_scc_charging                 1               True - 1/False - 0
is_scc_ok                       1               True - 1/False - 0
load_percentage                 002             %
max_ac_charger_current          50              A
max_charger_current             080             A
max_charger_range               080             A
output_mode                     single machine
parallel_instance_number        valid
pv_input_current                00              A
pv_input_voltage                222.6           V
serial_number                   92931903103094
total_ac_output_apparent_power  00115           VA
total_ac_output_percentage      001             %
total_charging_current          000             A
total_output_active_power       00099           W
work_mode                       Battery Mode
```
Cheers!

**Comment by dricote** [comment]
as a troubleshooting, tell you, when installing grafana this not found:

Start grafana and set to run at boot
```
sudo systemctl unmask grafana
sudo systemctl start grafana
sudo systemctl enable grafana
```
In the grafana installation it indicates the following:

```
### NOT starting on installation, please execute the following statements to configure grafana to start automatically using systemd
 sudo /bin/systemctl daemon-reload
 sudo /bin/systemctl enable grafana-server
### You can start grafana-server by executing
 sudo /bin/systemctl start grafana-server
```

I thought it interesting to tell you

Cheers!

**Comment by jblance** [comment]
After installing the missing package it looks to be working. Do you get info in the mosquito_sub window from the service?
If not it is likely that you need to configure you system to allow your user access to /dev/hidraw0

**Comment by dricote** [comment]
I have decided to run the daemon as root and with this it already works

`mv /etc/systemd/user/mpp-solar.service /etc/systemd/system/mpp-solar.service`

**Comment by jblance** [comment]
ok, thats another way to do it
Sounds like this issue is now resolved
